qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteCompletionInExpressionWithSwitchWhenSectionTest.java

```diff
@@ -0,0 +1,95 @@
+*******************************************************************************/
+package com.redhat.qute.services.completions;
+
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for Qute completion in expression for #switch and #when section.
+ *
+ */
+public class QuteCompletionInExpressionWithSwitchWhenSectionTest {
+
+	@Test
+	public void propertyPartWhenNoExisting() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is | }\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("ON : MachineStatus", "ON", r(2, 7, 2, 7)), //
+				c("OFF : MachineStatus", "OFF", r(2, 7, 2, 7)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(2, 7, 2, 7)));
+	}
+	
+	@Test
+	public void propertyPartWhenAddtionalSameLine() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is ON |}\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("OFF : MachineStatus", "OFF", r(2, 10, 2, 10)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(2, 10, 2, 10)));
+	}
+	
+	@Test
+	public void propertyPartWhenNewLine() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is ON }\r\n" + //
+				"  {#is | }\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("OFF : MachineStatus", "OFF", r(3, 7, 3, 7)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(3, 7, 3, 7)));
+	}
+	
+	@Test
+	public void propertyPartWhenWithIn() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is ON }\r\n" + //
+				"  {#is in BROKEN |}\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("OFF : MachineStatus", "OFF", r(3, 17, 3, 17)));
+	}
+	
+	@Test
+	public void propertyPartWhenWithInEnumName() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#when Machine.status}\r\n" + //
+				"  {#is ON }\r\n" + //
+				"  {#is in |}\r\n" + // <-- completion here
+				"{/when}";
+		testCompletionFor(template, //
+				c("BROKEN : MachineStatus", "BROKEN", r(3, 10, 3, 10)), //
+				c("OFF : MachineStatus", "OFF", r(3, 10, 3, 10)));
+	}
+	
+	@Test
+	public void propertyPartSwitchNoExisting() throws Exception {
+		String template = "{@org.acme.Machine Machine}\r\n" + //
+				"{#switch Machine.status}\r\n" + //
+				"  {#case | }\r\n" + // <-- completion here
+				"{/switch}";
+		testCompletionFor(template, //
+				c("ON : MachineStatus", "ON", r(2, 9, 2, 9)), //
+				c("OFF : MachineStatus", "OFF", r(2, 9, 2, 9)), //
+				c("BROKEN : MachineStatus", "BROKEN", r(2, 9, 2, 9)));
+	}
```

Comment:
Please write test with no completion (to check that no exception are thrown) with following usecases:

```
{#when}
  {#is | }
{/when}
```

```
{#when XXX}
  {#is | }
{/when}
```

```
{@org.acme.Item item}
{#when item.name}
  {#is | }
{/when}
```

```  
{#is | }
```